src/client/Microsoft.Identity.Client/AppConfig/ApplicationConfiguration.cs

```diff
@@ -107,24 +107,27 @@ public string ClientVersion
         public string Component { get; internal set; }
         public IDictionary<string, string> ExtraQueryParameters { get; internal set; } = new Dictionary<string, string>();
         public bool UseRecommendedDefaultRedirectUri { get; internal set; }
-
         public bool ExperimentalFeaturesEnabled { get; set; } = false;
-
         public IEnumerable<string> ClientCapabilities { get; set; }
         public bool SendX5C { get; internal set; } = false;
         public bool LegacyCacheCompatibilityEnabled { get; internal set; } = true;
         public bool CacheSynchronizationEnabled { get; internal set; } = true;
         public bool MultiCloudSupportEnabled { get; set; } = false;
-
         public bool RetryOnServerErrors { get; set; } = true;
 
-        public ManagedIdentityId ManagedIdentityId { get; internal set; }
+        public Func<AppTokenProviderParameters, Task<AppTokenProviderResult>> AppTokenProvider;
 
+        #region ManagedIdentity
+        public ManagedIdentityId ManagedIdentityId { get; internal set; }
         public bool IsManagedIdentity { get; }
 
-        public Func<AppTokenProviderParameters, Task<AppTokenProviderResult>> AppTokenProvider;
+        public CryptoKeyType ManagedIdentityCredentialKeyType { get; internal set; }

```

Comment:
Why do you need these 2 variables here instead of single object for KeyMaterialManager which have both the cert and CryptoKeyType?